### DISCUSSION TITLE
conflicts between libelf and elfutils

### DISCUSSION THREAD

**Discussion by byxrolland** [Homebrew/discussions#2057] [Linux]
This two packages both want to link lib/libelf.a. I don't know what to do,

**Comment by cho-m** [comment]
On Linux, the old, unmaintained `libelf` formula is being replaced with maintained `elfutils`.

`libelf` has been made keg-only, but Homebrew might not unlink an already installed copy, so you may need to run
```
brew unlink libelf
```

**EDIT:** At some future date, `libelf` should no longer be needed as dependency on Linux and should be safe to uninstall. I am working on this for:
- `llvm@7`, `llvm@8` - https://github.com/Homebrew/homebrew-core/pull/83787
- `llvm@9` - https://github.com/Homebrew/homebrew-core/pull/83780

The main `llvm` formula will be modified in one of the other open `llvm` PRs. I'll be creating a PR for remaining `avrdude` formula soon.

**Reply by byxrolland** [reply]
On my machine, it is still needed for gnuplot, llvm, matplotplusplus, mesa, mesa-glu and qt@5. Therefore I can't uninstall it.

**Reply by cho-m** [reply]
That is currently due to `llvm`. Once `llvm` is changed from `libelf` to `elfutils`, it should no longer be needed. The plan is to do this modification in the existing `llvm` PR https://github.com/Homebrew/homebrew-core/pull/83017

What you are seeing is the full dependency tree, i.e. 
- `matplotplusplus -> gnuplot -> qt@5 -> mesa -> llvm`
- `mesa-glu -> mesa -> llvm`

**Reply by byxrolland** [reply]
OK, Thanks.
